sdk/search/ai-search/review/ai-search.api.md

```diff
@@ -0,0 +1,285 @@
+            };
+        };
+    };
+}
+
+// @public (undocumented)
+export interface GetDocumentOptions extends OperationOptions {
+    selectedFields?: string[];
+}
+
+// @public
+export interface IndexAction {
+    [property: string]: any;
+    actionType?: IndexActionType;
+}
+
+// @public
+export type IndexActionType = 'upload' | 'merge' | 'mergeOrUpload' | 'delete';
+
+// @public
+export interface IndexDocumentsResult {
+    readonly results?: IndexingResult[];
+}
+
+// @public
+export interface IndexingResult {
+    readonly errorMessage?: string;
+    readonly key?: string;
+    readonly statusCode?: number;
+    readonly succeeded?: boolean;
+}
+
+// @public (undocumented)
+export type KnownKeys<T> = {
+    [K in keyof T]: string extends K ? never : number extends K ? never : K;
+} extends {
+    [_ in keyof T]: infer U;
+} ? U : never;
+
+// @public (undocumented)
+export interface ListSearchResultsPageSettings {
+    nextLink?: string;
+    nextPageParameters?: SearchRequest;
+}
+
+// @public (undocumented)
+export interface ModifyIndexOptions extends OperationOptions {
+    throwOnAnyFailure?: boolean;
+}
+
+// @public (undocumented)
+export function odata(strings: TemplateStringsArray, ...values: unknown[]): string;
+
+// @public
+export type QueryType = 'simple' | 'full';
+
+// @public (undocumented)
+export type ReplaceProperties<T, From, To> = {
+    [K in keyof T]: K extends keyof From ? T[K] extends From[K] ? K extends keyof To ? To[K] : T[K] : T[K] : T[K];
+};
+
+// @public
+export class SearchApiKeyCredential implements ServiceClientCredentials {
+    constructor(apiKey: string);
+    signRequest(webResource: WebResource): Promise<WebResource>;
+    updateKey(apiKey: string): void;
+}
+
+// @public (undocumented)
+export type SearchDocumentsResult<T> = ReplaceProperties<GeneratedSearchDocumentsResult, {
+    readonly results?: GeneratedSearchResult[];
+}, {
+    readonly results?: Array<SearchResult<T>>;
+}>;
+
+// @public (undocumented)
+export class SearchIndexClient<T> {
+    constructor(endpoint: string, indexName: string, credential: SearchApiKeyCredential, options?: SearchIndexClientOptions);
+    readonly apiVersion: string;
+    // (undocumented)
+    autocomplete(suggesterName: string, searchText: string, options?: AutocompleteOptions): Promise<AutocompleteResult>;
+    // (undocumented)
+    count(options?: CountOptions): Promise<number>;
+    // (undocumented)
+    deleteDocuments(keyName: string, keyValues: string[], options?: DeleteDocumentsOptions): Promise<IndexDocumentsResult>;
+    readonly endpoint: string;
+    // (undocumented)
+    getDocument(key: string, options?: GetDocumentOptions): Promise<T>;
+    readonly indexName: string;
+    // (undocumented)
+    listSearchResults<Fields extends keyof T>(searchText: string, options?: SearchOptions<T, Fields>): PagedAsyncIterableIterator<SearchResult<Pick<T, Fields>>, SearchDocumentsResult<Pick<T, Fields>>, ListSearchResultsPageSettings>;
```

Comment:
Double check how the docs render - it may be worth aliasing these return types with a good name so docs render better.